File: CSharp/StrategyCSharp/Shape.cs
using Moq;

namespace StrategyCSharp
{
    public class Shape
    {
        private readonly Type _type;

        public Shape(Type type)
        {
            _type = type;
        }

        public void Draw(ICanvas canvas)
        {
            canvas.DrawVertex(new Point(0, 0));
            canvas.DrawVertex(new Point(2, 0));
            if (_type == Type.Triangle)
            {
                canvas.DrawVertex(new Point(1, 2));
            }
            else
            {
                canvas.DrawVertex(new Point(0, 2));
                canvas.DrawVertex(new Point(2, 2));
            }

            if (_type == Type.Pentagon)
            {
                canvas.DrawVertex(new Point(1, 3));
            }
            else if (_type == Type.Hexagon)
            {
                canvas.DrawVertex(new Point(3, 1));
                canvas.DrawVertex(new Point(-1, 1));
            }
        }
    }
}

File: cppVS/Strategy/Shape.cpp
#include "pch.h"
#include "Shape.h"

Shape::Shape(const Type t) : m_type(t) {}

void Shape::Draw(ICanvas& canvas) const
{
    canvas.DrawVertex({ 0, 0 });
    canvas.DrawVertex({ 2, 0 });
    if (m_type == Type::Triangle) {
        canvas.DrawVertex({ 1, 2 });
    }
    else {
        canvas.DrawVertex({ 0, 2 });
        canvas.DrawVertex({ 2, 2 });
    }
    if (m_type == Type::Pentagon) {
        canvas.DrawVertex({ 1, 3 });
    }
    else if (m_type == Type::Hexagon)
    {
        canvas.DrawVertex({ 3, 1 });
        canvas.DrawVertex({ -1, 1 });
    }
}

